client/lib/analytics/utils/get-tracking-prefs.ts

```diff
@@ -14,40 +14,47 @@ export type TrackingPrefs = {
 	};
 };
 
-const allBucketsFalse: TrackingPrefs[ 'buckets' ] = {
-	essential: false,
-	analytics: false,
-	advertising: false,
+const prefsDisallowAll: TrackingPrefs = {
+	ok: false,
+	buckets: {
+		essential: true, // essential bucket is always allowed
+		analytics: false,
+		advertising: false,
+	},
 };
 
-const allBucketsTrue: TrackingPrefs[ 'buckets' ] = {
-	essential: true,
-	analytics: true,
-	advertising: true,
+const prefsAllowAll: TrackingPrefs = {
+	ok: true,
+	buckets: {
+		essential: true,
+		analytics: true,
+		advertising: true,
+	},
 };
 
-export const parseTrackingPrefs = ( cookieV2?: string, cookieV1?: string ): TrackingPrefs => {
+export const parseTrackingPrefs = (
+	cookieV2?: string,
+	cookieV1?: string,
+	defaultPrefs = prefsDisallowAll
+): TrackingPrefs => {
 	const { ok, buckets }: Partial< TrackingPrefs > = cookieV2 ? JSON.parse( cookieV2 ) : {};
 
 	if ( typeof ok === 'boolean' ) {
 		return {
 			ok,
 			buckets: {
-				...allBucketsFalse,
+				...defaultPrefs.buckets,
 				...buckets,
 			},
 		};
 	} else if ( cookieV1 && [ 'yes', 'no' ].includes( cookieV1 ) ) {
 		return {
+			...prefsAllowAll,
 			ok: cookieV1 === 'yes',
-			buckets: allBucketsTrue,
```

Comment:
Opinion: The spread operator here might be needlessly fancy. Since we are checking that cookieV1 exists and that it is either yes or no, `ok` will be overridden every time. I think `buckets: prefsAllowAll.buckets` would be clearer when reading the code personally

Reply:
Coolio, that does make sense, better to be explicit.